src/plsql/flow_rest_auth.pkb

```diff
@@ -0,0 +1,154 @@
+create or replace package body flow_rest_auth
+as
+
+  function get_rest_roles
+    return flow_rest_roles_nt
+    pipelined
+  as
+    l_roles apex_application_global.vc_arr2;  
+  begin
+
+    l_roles(1) := flow_rest_constants.c_rest_role_read;
+    l_roles(2) := flow_rest_constants.c_rest_role_write;
+    l_roles(3) := flow_rest_constants.c_rest_role_admin;
+
+    for rec_roles in (select id, name 
+                        from user_ords_roles 
+                        where name in (select column_value from table(l_roles)))
+    loop
+      PIPE ROW (flow_rest_role_ot ( rec_roles.id, rec_roles.name ));
```

Comment:
Pipelined table function does not need Schema level types.
Types can be declared in the package header..